compiler/rustc_codegen_llvm/src/llvm/diagnostic.rs

```diff
@@ -19,18 +18,17 @@ pub enum OptimizationDiagnosticKind {
     OptimizationRemarkOther,
 }
 
-pub struct OptimizationDiagnostic<'ll> {
+pub struct OptimizationDiagnostic {
     pub kind: OptimizationDiagnosticKind,
     pub pass_name: String,
-    pub function: &'ll Value,
```

Comment:
This field should be kept, even if it's currently unused on the Rust side, because it reflects part of a larger LLVM data structure.

(Keeping C++ and Rust in sync is easier if we don't have ad-hoc gaps in individual structures and enums.)